parse_nucleotide_fasta.c

```diff
@@ -0,0 +1,263 @@
+        {"match", required_argument, NULL, 'm'},
+        {"help", no_argument, NULL, 'h'},
+        {NULL, 0, NULL, 0}
+        /*
+        * Other future arguments:
+        *   -out <output_file>
+        *   -v <verbose_output>
+        *   -seq <sequence_number> // only searches for the following sequences in the file (marked by >)
+        *       example: -seq 2 would only work with the data chunk after the second > but before the third >
+        *       -seq 2,4 should be allowed to be entered
+        */
+    };
+
+    int opts;
+    int argument_legality = -1;
+    while ((opts = getopt_long_only(argcount, argvalues, "f:c::k:m:h", long_options, NULL)) != -1) {
+        switch (opts) {
+            case 'f':
+                argument_legality = 0;
+                flags[0].flag_raised = 1;
+                strcpy(flags[0].flag_value, optarg);
+                printf("File %s passed in.\n", optarg);
+                break;
+            case 'c':
+                argument_legality = 0;
+                flags[1].flag_raised = 1;
+                if (optarg == NULL) {
+                    printf("Calculating global GC content.\n");
+                } else {
+                    strcpy(flags[1].flag_value, optarg);
+                    printf("Calculating GC content per %s nucleotides\n", optarg);
+                }
+                break;
+            case 'k':
+                argument_legality = 0;
+                flags[2].flag_raised = 1;
+                strcpy(flags[2].flag_value, optarg);
+                printf("Determining all k-mers of size %s.\n", optarg);
+                break;
+            case 'm':
+                argument_legality = 0;
+                flags[3].flag_raised = 1;
+                strcpy(flags[3].flag_value, optarg);
+                printf("Searching for all instances of %s in file.\n", optarg);
+                break;
+            case 'h':
+                printf("Parse_nuc_fasta Options: \n");
+                printf("\t-help : displays this message.\n");
+                printf("\t-file <file> : Directs the program to the fasta file.\n");
+                printf("\t-gc : global gc count.\n");
+                printf("\t-gc=<x> : gc count per x bases.\n");
+                printf("\t-k_mers <k> : lists all k-mers of size k.\n");
+                printf("\t-match <string> : searches for all instances of a string. (only words for same case searches)\n");
+                argument_legality = -1;
+                break;
+            case '?':
+                printf("Unknown or incorrect argument entered, exiting program. Please see maunal or enter ./parse_nuc_fasta -help\n");
+                argument_legality = -1;
+                break;
+            default:
+                printf("No legal options entered. Please see maunal or enter ./parse_nuc_fasta -help\n");
+                argument_legality = -1;
+                break;
+        }
+    }
+    if (flags[0].flag_raised == 0) {
```

Comment:
Again, if you move to using bool, this statement simplifies to just `if (flags[0].flag_raised) {`